cmd/oras/repository/ls.go

```diff
@@ -67,9 +78,22 @@ func listRepository(opts repositoryOptions) error {
 		return err
 	}
 	return reg.Repositories(ctx, opts.last, func(repos []string) error {
-		for _, repo := range repos {
-			fmt.Println(repo)
+		for _, repopath := range repos {
+			if inNamespace(repopath, opts.namespace) {
+				fmt.Println(repopath)
+			}
 		}
 		return nil
 	})
 }
+
+func inNamespace(repopath string, requestedNamespace string) bool {
+	if requestedNamespace == "" {
+		return true
+	}
+	before, after, found := strings.Cut(repopath, requestedNamespace)
+	if !found || after == "" || before != "" {
+		return false
+	}
+	return after[0] == '/'
```

Comment:
It partially works and is not readable. Can we choose a simpler implementation?

The reason why it partially works is that the `inNamespace("foo", "foo")` returns `false`, which should be `true`. You need unit tests to ensure code quality.

Reply:
It seems to be a UX issue other than a bug.

Reply:
I need advice on a readable and simpler implementation.

Reply:
Used `strings.HasPrefix(repopath, opts.namespace+"/") `here after discussion.